### ISSUE TITLE
`_QueryParamsLikelyByMistake & QueryParams` Type Issue

### ISSUE THREAD

**Issue by mitchuman** [sanity-io/next-sanity#941]
**Describe the bug**

Weird error logged which is haulting deployments on v7.0.14

Logs:
```ts
./src/lib/sanity.ts:23:32
Type error: Argument of type "QueryParams | undefined" is not assignable to parameter of type "_QueryParamsLikelyByMistake & QueryParams".
  Type "undefined" is not assignable to type "_QueryParamsLikelyByMistake & QueryParams".
    Type "undefined" is not assignable to type "_QueryParamsLikelyByMistake".
  21 | 	} = {},
  22 | ) {
> 23 | 	return client.fetch<T>(query, params, {
     | 	                              ^
  24 | 		...(dev && {
  25 | 			token: process.env.NEXT_PUBLIC_SANITY_TOKEN,
  26 | 		}),
error Command failed with exit code 1. (https://ntl.fyi/exit-code-1)
```

**To Reproduce**

```ts
// ./src/lib/sanity.ts

export function fetchSanity<T>(
  query: string,
  {
    params,
    tags,
  }: {
    params?: QueryParams
    tags?: string[]
  } = {},
) {
  return client.fetch<T>(query, params, {
    ...(dev && {
      token: process.env.NEXT_PUBLIC_SANITY_TOKEN,
    }),
    next: {
      revalidate: dev ? 0 : false,
      tags,
    },
  })
}
```

**Expected behavior**

No issue happening on v7.0.10 (have not tested on versions higher than this other than v7.0.14).
Issue also only happens on production deployments on Netlify (have not tried on Vercel). No issue on local dev or local build.

**Screenshots**

**Which versions of Sanity are you using?**

```
@sanity/cli (global)             3.26.0 (latest: 3.26.1)
@sanity/dashboard                 3.1.6 (up to date)
@sanity/eslint-config-studio      3.0.1 (up to date)
@sanity/orderable-document-list   1.2.0 (up to date)
@sanity/vision                   3.26.0 (latest: 3.26.1)
sanity                           3.26.0 (latest: 3.26.1)
```

**What operating system are you using?**

**Which versions of Node.js / npm are you running?**

```
9.6.7
v18.17.1
```

**Additional context**

**Security issue?**

Any security issues should be submitted directly to [security@sanity.io](mailto:security@sanity.io). In order to determine whether you are dealing with a security issue, ask yourself these two questions:

- Can I access something that's not mine, or something I shouldn't have access to?
- Can I disable something for other people? If the answer to either of those two questions are "yes", then you're probably dealing with a security issue. Note that even if you answer "no" to both questions, you may still be dealing with a security issue, so if you're unsure, just email us at [security@sanity.io](mailto:security@sanity.io.

**Comment by mitchuman** [comment]
Looks like this is now fixed in v7.0.15 with the type `QueryParamsWithoutQueryOptions`

**Comment by stipsan** [comment]
Sounds like the issue where duplicate instances of `@sanity/client` in your `node_modules` :) closing
